File: ruby/5-loops.rb
#!/bin/ruby
counter=0

while counter < 10
	puts "counter: #{counter}"
	counter+=1
end

until counter == 20
	puts "counter: #{counter}"
	counter+=1
end

until  counter == 30
	if counter == 25 
		puts "I don't like 25"
		counter+=1
		next
	elseif
		puts "counter: #{counter}"
	end
	counter+=1
end

while true
	if counter == 35
		puts "enough with counting"
		break;
	end
	puts "counter: #{counter}"
	counter+=1
end

(1..5).each do | i|
  puts "1..5 -> i=#{i}"
end

for l in (1...5)
  puts "for 1...5 -> l=#{l}"
  case 
  when l == 3
    puts "l=3"
  end
end

5.times do |a|
  puts "5.times a=#{a}"
end

number=["one", "two", "tree"]
number.each do | l |
	puts l
end

for f in number
  puts "f=#{f}"
  case f
  when "two"
    puts "it's two"
  else
    puts "something else than two"
  end
end

File: c/5-loops.c
#include <stdio.h>

int main()
{
	int b, c, d, e, f, g=0;
	for (int i=0; i < 10; i++)
	{
		printf("i=%d\n", i);
		b=i;
	}

	printf("start: b=i=%d\n", b);
	for (; b < 20; )
		printf("b++=%d\n", b++);
	
	printf("start: c=b=%d\n", b);
	for (c=b; c < 30; )
		printf("++c=%d\n", ++c);
	
	printf("start: d=c, e=c=%d\n", c);
	for (d=c, e=c; d < 40; )
		printf("++d=%d e++=%d\n", ++d, e++);
	
	f=e;
	printf("start: f=e=%d\n", e);
	while (e < 50)
	{
		printf("++e=%d e--=%d f--=%d ++f=%d\n", ++e, e--, f--, ++f);
		g++;
		if (g > 10) break;
	}
	
	printf("start: g=%d\n", g);
	do {
		printf("g--=%d\n", g--);
	} while(g > 0 );


	for (int h=0; h < 10; h++)
	{
		if (h == 4) {
			puts("I've got 4");			
			continue;
		}
		if (h==5) goto jump;
		switch(h)
		{
			case 1: puts("one"); break;
					default: printf("something other than 1, because %d\n", h); break;
					}
					continue;			
			jump:
					printf("side jump: %d\n", h);
	}

	return 0;
}	
